server/commands.go

```diff
@@ -0,0 +1,472 @@
+		Trigger:          commandIcebreakerShowProposals,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Show a list of proposed Icebreaker questions. Channel owners only",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerShowProposals))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerShowApproved,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Show the list of Icebreaker questions",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerShowApproved))
+	}
+	if err := p.API.RegisterCommand(&model.Command{
+		Trigger:          commandIcebreakerResetToDefault,
+		AutoComplete:     true,
+		AutoCompleteDesc: "Resets the Icebreaker questions to the default ones from this plugin",
+	}); err != nil {
+		return errors.Wrapf(err, fmt.Sprintf("Failed to register %s command", commandIcebreakerResetToDefault))
+	}
+	return nil
+}
+
+// ExecuteCommand executes a command that has been previously registered via the RegisterCommand
+// API.
+func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	trigger := strings.TrimPrefix(args.Command, "/")
+
+	if strings.HasPrefix(trigger, commandIcebreaker) {
+		if strings.HasPrefix(trigger, commandIcebreakerAdd) {
+			return p.executeCommandIcebreakerAdd(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerApprove) {
+			return p.executeCommandIcebreakerApprove(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerReject) {
+			return p.executeCommandIcebreakerReject(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerRemove) {
+			return p.executeCommandIcebreakerRemove(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerClearAllProposals) {
+			return p.executeCommandIcebreakerClearAllProposals(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerClearAllApproved) {
+			return p.executeCommandIcebreakerClearAllApproved(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerShowProposals) {
+			return p.executeCommandIcebreakerShowProposals(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerShowApproved) {
+			return p.executeCommandIcebreakerShowApproved(args), nil
+		} else if strings.HasPrefix(trigger, commandIcebreakerResetToDefault) {
+			return p.executeCommandIcebreakerResetToDefault(args), nil
+		} else {
+			return p.executeCommandIcebreaker(args), nil
+		}
+	} else {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         fmt.Sprintf("Unknown command: " + args.Command),
+		}, nil
+	}
+}
+
+func (p *Plugin) executeCommandIcebreakerResetToDefault(args *model.CommandArgs) *model.CommandResponse {
+	sourceUser, _ := p.API.GetUser(args.UserId)
+	if !sourceUser.IsSystemAdmin() { //TODO: Check for Channel owner instead of System Admin
```

Comment:
This is a duplicate logic for multiple commands. This can be created as a util function -

```go
func RequireAdminUser(sourceUser *model.User) *model.CommandResponse {
        if !sourceUser.IsSystemAdmin() { //TODO: Check for Channel owner instead of System Admin
	       return &model.CommandResponse {
			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
			Text:         "You need to be admin in order to clear all proposed questions",
		}
	}

        return nil
}
```